web/ui/src/Health/serviceHealth.js

```diff
@@ -29,62 +29,49 @@
     function($translate){
 
         var statuses = {};
-        var serviceHealths = {};
 
         // updates health check data for all services
         function update(serviceList) {
 
-            var serviceHealthCheck, instanceHealthCheck,
-                serviceStatus, instanceStatus, instanceUniqueId,
-                instance, service;
+            var serviceStatus, instanceStatus, instanceUniqueId, service;
 
             statuses = {};
 
             // iterate services healthchecks
             for(var serviceId in serviceList){
                 service = serviceList[serviceId];
-                serviceHealthCheck = serviceHealths[serviceId];
                 serviceStatus = new Status(
                     serviceId,
                     service.name,
                     service.model.DesiredState);
 
-                // if no healthcheck for this service mark as not running
-                if(!serviceHealthCheck){
-                    serviceStatus.statusRollup.incNotRunning();
-                    serviceStatus.evaluateStatus();
-
-                // otherwise, look for instances
-                } else {
-
-                    // iterate instances healthchecks
-                    for(var instanceId in serviceHealthCheck){
-                        instanceHealthCheck = serviceHealthCheck[instanceId];
-                        instanceUniqueId = serviceId +"."+ instanceId;
-                        // evaluate the status of this instance
-                        instanceStatus = new Status(
-                            instanceUniqueId,
-                            service.name +" "+ instanceId,
-                            service.model.DesiredState);
-
-                        instanceStatus.evaluateHealthChecks(instanceHealthCheck);
-
-                        // attach status to instance
-                        instance = service.instances.find(instance => instance.id === instanceId);
-                        if(instance){
-                            instance.status = instanceStatus;
-                        }
-
-                        // add this guy's statuses to hash map for easy lookup
-                        statuses[instanceUniqueId] = instanceStatus;
-                        // add this guy's status to his parent
-                        serviceStatus.children.push(instanceStatus);
-                    }
+                // refresh list of instances
+                service.getServiceInstances();
 
-                    // now that this services instances have been evaluated,
-                    // evaluate the status of this service
-                    serviceStatus.evaluateChildren();
-                }
```

Comment:
This performs roughly the same as the deleted code above, except it operates on the instances health checks directly instead of a separate health check dictionary